src/cascadia/TerminalSettingsModel/AppearanceConfig.cpp

```diff
@@ -50,6 +54,25 @@ Json::Value AppearanceConfig::ToJson() const
     JsonUtils::SetValueForKey(json, SelectionBackgroundKey, _SelectionBackground);
     JsonUtils::SetValueForKey(json, CursorColorKey, _CursorColor);
     JsonUtils::SetValueForKey(json, OpacityKey, _Opacity, JsonUtils::OptionalConverter<double, IntAsFloatPercentConversionTrait>{});
+    if (HasDarkColorSchemeName() && HasLightColorSchemeName())
+    {
+        // check if the setting is coming from the UI, if so grab the ColorSchemeName until the settings UI is fixed..
+        if (_ColorSchemeName != _DarkColorSchemeName)

```

Comment:
Is this talking about line 57 or 60?

Reply:
Line 60.

Reply:
In that case, it would change the logic. We are checking here if the ColorSchemeName differs. This is because ColorSchemeName is still being used by the UI. We check that value differs, indicating the change came from the UI. We have to then can choose to use the old variable for saving to the JSON.